src/natcap/invest/utils.py

```diff
@@ -538,6 +541,30 @@ def build_lookup_from_csv(
     return lookup_dict 
 
 
+def read_csv_to_dataframe(path, to_lower=True, strip=True, **kwargs):
+    """
+    Wrapper around ``pandas.read_csv`` that standardizes the column names by
+    stripping leading/trailing whitespace and making all lowercase.
+    This helps avoid common errors caused by user-supplied CSV files with
+    column names that don't exactly match the specification.
+
+    Args:
+        path (string): path to a CSV file
+        to_lower (bool): if True, convert all column names to lowercase
+        strip (bool): if True, strip leading/trailing whitespace from all
+            column names
+        **kwargs: any kwargs that are valid for ``pandas.read_csv`` or
+            ``pandas.read_excel``
+    """
+    dataframe = pandas.read_csv(path, **kwargs)
```

Comment:
One of the critical things that `build_lookup_from_csv` does is checking for the presence of a BOM, which then changes the encoding that we assume.  I'd suggest that we include that checking here as well BUT, the presence of `**kwargs` makes that a little more interesting because a user should also be able to provide the `encoding` keyword argument and have that take precedent.

Reply:
Do you mean user, as in where `read_csv_to_dataframe` is called, or as in the actual user supplying the files? I don't think there's currently the option for them to supply a specific encoding

Reply:
In this case, "user" is the caller of the function.  You're right that there isn't currently an option to provide a specific encoding to `utils.read_csv_to_dataframe` (and that's probably OK for our purposes until we know otherwise).  But the thing is that this proposed function is accepting _all possible keyword arguments_ that are accepted to the `pandas.read_csv`, and so if the caller provides an `encoding` kwarg (which is a valid input to `pandas.read_csv()`), we should honor that.

The catch is that if the caller _doesn't_ provide an `encoding` kwarg, then we should take advantage of our experience dealing with the CSV encodings we usually deal with and assume UTF-8, either with or without a BOM.  If we don't support an optional BOM we risk needing to spend more time debugging this on the forums down the road and needing to patch it later on.

So, the main question I have for you comes back to the use cases for this function: what are the features that we demonstrably need (given our 3+ examples of where this function will be used)?  What are the problems that this helper function will solve?  And, more to the point, is the `**kwargs` needed for our use cases?

Reply:
By including a kwarg `encoding='utf-8-sig'` in the method signature, I believe it should default to UTF-8 with/without a BOM, and still allow the caller to override it, see comment below about 'utf-8-sig'. 

There aren't any cases where stripping the whitespace is undesirable, so I don't think there needs to be a kwarg for that. Seems safe to assume it should always be stripped.

There are at least 3 cases where we don't want to make column names lowercase, at least until we decide if that should be a convention. So I'm making that a kwarg `to_lower=False`.

There are 5 cases where the `**kwargs` is used: see changes in recmodel_client.py, globio.py, validation.py, wave_energy.py, wind_energy.py. So that makes sense to me to include as a feature of the helper function.